### PR TITLE
fix(core): generate UUID primary-key setter in init_superuser

### LINKED ISSUE #4237: Bug: `createsuperuser` writes the nil UUID (`00000000-0000-0000-0000-000000000000`) into the user PK
## Summary

`manage createsuperuser` produces an `auth_users` row whose `id` (UUID PK) is the nil UUID `00000000-0000-0000-0000-000000000000` instead of a freshly generated v4 (or v7) UUID. This means **only one** superuser can ever exist; a second `createsuperuser` invocation will fail with a unique-constraint violation on the PK.

## Reproduction

Fresh Postgres + migrations applied. Then:

```bash
cargo run --bin manage -- createsuperuser
# Username: admin
# Email address: admin@example.com
# Password: adminpass1
# Password (again): adminpass1
# Create superuser? [Y/n] Y
# Superuser created successfully!
```

Inspect the row:

```sql
SELECT id, username, email, is_superuser FROM auth_users;
```

```
                  id                  | username |       email       | is_superuser
--------------------------------------+----------+-------------------+--------------
 00000000-0000-0000-0000-000000000000 | admin    | admin@example.com | t
```

`username` and `email` have `UNIQUE` constraints, so the duplicate-username angle is hidden until you try to create a *second* superuser, at which point the PK collision will surface as well.

## Expected

`id` should be `Uuid::new_v4()` (or whichever default the `#[user]` `#[model]` derive uses for other rows). Other tables created by the same migrations get random UUIDs as expected; only the superuser created by this command appears to get the nil UUID.

## Suspected cause

Probable miss in the createsuperuser model construction path — the UUID default isn't being applied when the model is built outside the normal request/ORM-create flow. (Could be the same code path that #3187 \"auto-register SuperuserCreator for #[user] + #[model]\" touched.)

## Impact

- Any environment that needs more than one superuser (which is most non-toy deployments) hits a hard PK collision on the second `createsuperuser`.
- Even a single-superuser setup has a very surprising row-id that breaks anything joining/comparing UUIDs by integrity.

## Tracking

reinhardt-cloud tracking issue: [kent8192/reinhardt-cloud#587](https://github.com/kent8192/reinhardt-cloud/issues/587)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by kent8192**
## Summary

- Make the `#[user(full = true)] #[model] ...` macro pair generate a fresh `Uuid::now_v7()` assignment for the primary key inside `SuperuserInit::init_superuser`, immediately after the `Self::default()` call.
- Add an integration regression test that pins down both the non-nil-UUID and distinct-UUID-across-calls invariants.

## Type of Change

- Bug fix (non-breaking)

## Motivation and Context

Issue #4237 reports that `manage createsuperuser` writes the nil UUID `00000000-0000-0000-0000-000000000000` into the `auth_users` PK. The first invocation looks successful — the row lands cleanly — but the second `createsuperuser` collides on the unique primary key, so any environment that needs more than one superuser is blocked. Tracing the macro output, `generate_superuser_init_impl` builds the row via `let mut user = Self::default();` and never re-seeds the PK, so a `Uuid::default()` (= `Uuid::nil()`) is what reaches the database.

The fix detects the PK type at macro-expansion time using a small local shape probe and emits `user.<pk> = ::uuid::Uuid::now_v7();` (or `Some(now_v7())` for `Option<Uuid>`) right after `Self::default()` runs, mirroring the v7 generation that the `#[model]` derive already injects into its `new()` constructor. Non-Uuid PKs are left untouched, so any existing user type with an integer auto-increment PK keeps compiling unchanged.

## How Was This Tested

- `cargo nextest run -p reinhardt-integration-tests --test auth -- superuser_init_integration`
- Workspace-wide: `cargo make fmt-check`, `cargo make clippy-check`

## Checklist

- [x] Code follows project style guidelines (`cargo make fmt-check`)
- [x] No new clippy warnings introduced (`cargo make clippy-check`)
- [x] All existing tests pass (`cargo nextest run`)
- [x] New integration regression test added in `tests/integration/tests/auth/superuser_init_integration.rs`
- [x] No new `todo!()` / `// TODO:` introduced
- [x] PR description follows `.github/PULL_REQUEST_TEMPLATE.md`

## Labels to Apply

- `bug`

## Related Issues

Fixes #4237

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/reinhardt-core/macros/src/user_attribute.rs [hunk 1]
```diff
@@ -318,6 +318,30 @@ fn generate_permissions_mixin_impl(
...
 
+/// Returns `(is_uuid, is_option)` for a Uuid-shaped primary-key field.
+///
+/// Mirrors the detection that `model_derive.rs::is_uuid_type` performs
+/// for `#[model]` PKs, kept local to avoid widening the visibility of a
+/// helper that is otherwise only used by the model derive itself. The
+/// `#[model]` derive injects `Uuid::now_v7()` into its generated `new()`
+/// constructor, but `init_superuser` builds the row via `Self::default()`,
+/// where `Uuid::default()` is `Uuid::nil()` — issue #4237. Detecting the
+/// shape here lets us inject the same fresh-UUID assignment.
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-core/macros/src/user_attribute.rs]: Addressed in 8668a6d86. The doc comment now spells out that `is_option` reflects only the outer `Option<T>` wrapper independent of the inner type, and that callers must branch on `is_uuid` first.

File: crates/reinhardt-core/macros/src/user_attribute.rs [hunk 2]
```diff
@@ -318,6 +318,30 @@ fn generate_permissions_mixin_impl(
...
+		&& let Some(last_segment) = type_path.path.segments.last()
+		&& last_segment.ident == "Option"
+		&& let syn::PathArguments::AngleBracketed(args) = &last_segment.arguments
+		&& let Some(syn::GenericArgument::Type(inner)) = args.args.first()
+	{
+		return (last_segment_is(inner, "Uuid"), true);
+	}
+	(last_segment_is(ty, "Uuid"), false)
+}
+
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-core/macros/src/user_attribute.rs]: Not addressed here; tracking in #4246. Rationale: extracting a shared `pub(crate)` helper is a refactor across both macro entrypoints and lies outside the `#4237` bug-fix unit (per WU-1: 1 PR = 1 fix pattern); the helpers also have different return shapes (`pk_uuid_shape` carries `is_option` while `is_uuid_type` returns `bool`), so unification needs its own design decision.
